sdk/cosmosdb/cosmos/src/queryExecutionContext/parallelQueryExecutionContextBase.ts

```diff
@@ -471,7 +471,7 @@ export abstract class ParallelQueryExecutionContextBase implements ExecutionCont
       sqlQuerySpec["query"] = rewrittenQuery;
     }
 
-    const options = JSON.parse(JSON.stringify(this.options));
+    const options = Object.assign(Object.create(Object.getPrototypeOf(this.options)), this.options);
```

Comment:
Unless `options` has a strange prototype, is this different from `const options = { ...options }`?

Reply:
the line gives an error Block-scoped variable 'options' used before its declaration. is that the syntax?

Reply:
I believe, he meant `const options = { ...this.options }`